opensfm/test/test_types.py

```diff
@@ -268,12 +927,246 @@ def _get_dual_camera():
+
+    # test the rotation matrix
+    assert np.allclose(pose.get_rotation_matrix(), R_cw)
+    assert np.allclose(pose.get_R_world_to_cam(), R_cw)
+    assert np.allclose(pose.get_R_cam_to_world(), R_cw.transpose())
+
+    pose = pygeometry.Pose()
+    R_wc = special_ortho_group.rvs(3)
+    t_wc = np.random.rand(3)
+    T_wc = np.vstack((np.column_stack((R_wc, t_wc)), np.array([0, 0, 0, 1])))
+    pose.set_from_cam_to_world(T_wc)
+    assert np.allclose(pose.get_cam_to_world(), T_wc)
+    assert np.allclose(pose.get_world_to_cam(), np.linalg.inv(T_wc))
+    assert np.allclose(pose.get_R_cam_to_world(), R_wc)
+    assert np.allclose(pose.get_R_world_to_cam(), R_wc.transpose())
+    assert np.allclose(pose.get_t_cam_to_world(), t_wc)
+    assert np.allclose(pose.get_t_world_to_cam(), -R_wc.transpose().dot(t_wc))
+
+    pose2 = pygeometry.Pose()
+    pose2.set_from_cam_to_world(R_wc, t_wc)
+    assert np.allclose(pose2.get_cam_to_world(), T_wc)
+    assert np.allclose(pose2.get_world_to_cam(), np.linalg.inv(T_wc))
+    assert np.allclose(pose2.get_origin(), pose.get_t_cam_to_world())
+
+    pose.translation = t_cw
+    assert np.allclose(pose.translation, t_cw)
+    assert np.allclose(pose.get_t_world_to_cam(), t_cw)
+
+
+def test_pose_minimal_representation():
+    p1 = pygeometry.Pose()
+    # Check identity pose
+    p1.set_from_world_to_cam(np.array([0, 0, 0]), np.array([0, 0, 0]))
+    assert np.allclose(p1.get_rotation_matrix(), np.eye(3))
+    assert np.allclose(p1.get_R_world_to_cam_min(), np.zeros((1, 3)))
+    assert np.allclose(p1.get_R_cam_to_world_min(), np.zeros((1, 3)))
+    assert np.allclose(p1.get_cam_to_world(), np.eye(4))
+    assert np.allclose(p1.get_world_to_cam(), np.eye(4))
+
+    R_wc = special_ortho_group.rvs(3)
+    t_wc = np.random.rand(3)
+    pose = pygeometry.Pose()
+    r_wc = Rotation.from_dcm(R_wc).as_rotvec()  # same as cv2.Rodrigues
+    pose.set_from_cam_to_world(R_wc, t_wc)
+    assert np.allclose(pose.get_R_cam_to_world_min(), r_wc)
+    pose.set_from_world_to_cam(R_wc, t_wc)
+    assert np.allclose(pose.get_R_world_to_cam_min(), r_wc)
+
+    R_wc = special_ortho_group.rvs(3)
+    t_wc = np.random.rand(3)
+    pose = pygeometry.Pose()
+    r_wc = Rotation.from_dcm(R_wc).as_rotvec()  # same as cv2.Rodrigues
+    pose.set_from_world_to_cam(r_wc, t_wc)
+    assert np.allclose(R_wc, pose.get_R_world_to_cam())
+    assert np.allclose(pose.get_R_world_to_cam_min(), r_wc)
+    assert np.allclose(pose.get_R_cam_to_world_min(), -r_wc)
+    assert np.allclose(R_wc.transpose(), pose.get_R_cam_to_world())
+
+    R_wc = special_ortho_group.rvs(3)
+    pose = pygeometry.Pose()
+    pose.set_rotation_matrix(R_wc)
+    assert np.allclose(R_wc, pose.get_R_world_to_cam())
+    r_wc = Rotation.from_dcm(R_wc).as_rotvec()
+    assert np.allclose(r_wc, pose.get_R_world_to_cam_min())
+
+    R_wc = special_ortho_group.rvs(3)
+    pose = pygeometry.Pose()
+    r_wc = Rotation.from_dcm(R_wc).as_rotvec()
+    pose.rotation = r_wc
+
```

Comment:
That would be great if we could split those tests.

Reply:
I just saw that there are a few redundant asserts and tests. I'll try to split the tests and maybe write a `helper` to avoid the many redundant asserts.